### ISSUE TITLE
Last call of @info seems not to be flushed

### ISSUE THREAD

**Issue by xukai92** [PhilipVinc/TensorBoardLogger.jl#53]
The last call of @info seems not to be flushed. Is this tested or I did something wrong?

**Comment by PhilipVinc** [comment]
Do you have an example?
It [should flush](https://github.com/PhilipVinc/TensorBoardLogger.jl/blob/6f39163b45e83bbc189dd5eff7d55793064ad6fd/src/event.jl#L42) after every event is serialized.

It's not tested, though. 
I should test it, that's true.

**Comment by xukai92** [comment]
The main codes are
```julia
@showprogress "Distilling" for i in 1:n_lessons
        s = score(knowledge)
        gs = Tracker.gradient(() -> s, ps)
        Tracker.update!(opt, ps, gs)
        @info "train" score=Flux.data(s)
    end
    
acc = mean(onecold(ds.student(get_Z(ds, knowledge))) .== onecold(get_probability(knowledge)))
@info "train" finish_msg="Distillation with $acc towards knowledge." log_step_increment=0
```

The last `@info` never shows in my TFBoard. Is it related to `log_step_increment=0` I set?

**Comment by PhilipVinc** [comment]
Hmm. I cannot reproduce this on MacOs with tensor board 1.10.0.

Could you check that it is not TensorBoard that is not refreshing for some reason?
Simply start TensorBoard _after_ you have logged that last message. Do you see it?

Alternatively, if you checkout the branch `read_serialized` you can use the (un-exported) command `map_events`.
It's not really well documented and tested yet, but you should be able to extract the last event and see what's inside.

**Comment by PhilipVinc** [comment]
You could try running the following
```julia
tbl = TBLogger()
# do your stuff

events = []
TensorBoardLogger.map_events(x->push!(events,x), TensorBoardLogger.logdir(tbl))
display(last(events))
display(last(events).summary.value)
```
what's the output?

**Comment by xukai92** [comment]
> Simply start TensorBoard after you have logged that last message. Do you see it?

In fact this works! What exactly happens here?

**Comment by PhilipVinc** [comment]
Eh. I have no idea. 
I have noticed that sometimes TensorBoard does not refresh (even if you push the refresh button). 
I only saw it happening when you log subsequent messages with the same step.

It's possible that we do something that tensor board does not like.
But if they don't document how we should talk to TB, I can only take guesses.

**Comment by xukai92** [comment]
All right. I guess I will close this issue for now in thie case.

**Comment by PhilipVinc** [comment]
By the way, this [seems](https://github.com/tensorflow/tensorboard/issues/867) [known](https://github.com/pytorch/pytorch/issues/20477).
Seems it was fixed in v1.15
